### PR TITLE
Added documentation

### PR DESCRIPTION
**Pull request by daviroo**
Myself and @nicolasburtey added some docs for the graphql.  A long way to go but it's a start.  There are some questions we had while writing the docs which I have marked with `TODO`.

### CODE REVIEW

File: src/graphql/admin/schema.graphql [hunk 1]
```diff
@@ -237,10 +257,26 @@ type BTCWallet implements Wallet {
 
 type IntraLedgerTransaction implements Transaction {
   id: ID!
+
+  """
+  From which protocol the payment has been initiated.
```
Comment (bodymindarts) [inline src/graphql/admin/schema.graphql:262]: is `protocol` really the right name?

File: src/graphql/types/object/wallet-contact.ts [hunk 2]
```diff
@@ -9,11 +9,12 @@ import * as Accounts from "@app/accounts"
...
 import { checkedToUsername } from "@domain/users"
 
+//FIXME: Discrepancy with filename
 const UserContact = new GT.Object({
   name: "UserContact",
   fields: () => ({
-    username: { type: GT.NonNull(Username) },
-    alias: { type: ContactAlias },
+    username: { type: GT.NonNull(Username), description: "Actual identifier of the contact." },
+    alias: { type: ContactAlias, description: "Alias the user can set for this contact.\nOnly the user can see the alias attached to their contact." },
```
Comment (bodymindarts) [inline src/graphql/types/object/wallet-contact.ts]: This line is a bit long - shouldn't get by linting or?
Isn't there a better way to do multi-line string that respects our line width settings?

File: src/graphql/types/object/user.ts [hunk 3]
```diff
@@ -15,15 +15,17 @@ import { UnknownClientError } from "@core/error"
...
 const mainUserFields = () => ({
   id: { type: GT.NonNullID },
-  phone: { type: GT.NonNull(Phone) },
-  username: { type: Username },
+  phone: { type: GT.NonNull(Phone), description: "Phone number with international calling code." },
+  username: { type: Username, description: "Optional immutable user friendly identifier." },
   language: {
     type: GT.NonNull(Language),
     resolve: (source) => source.language,
+    description: "Preferred language for user.\nWhen value is 'default' the intent is to use preferred language from OS settings."
```
Comment (bodymindarts) [inline src/graphql/types/object/user.ts]: Can we use multiline string instead of `\n`?

File: src/graphql/admin/schema.graphql (hunk 1 above)
Comment (nicolasburtey) [reply src/graphql/admin/schema.graphql:262]: do you have a better name for describing onchain/lightning/intraledger?

File: src/graphql/types/abstract/transaction.ts [hunk 4]
```diff
@@ -34,10 +39,14 @@ export const transactionInterfaceFields = {
       return {
         formattedAmount: settlementUsdPerSatInCents.toString(),
         base: Math.round(settlementUsdPerSatInCents * 10 ** 4),
+        //TODO: What is offset?  What is difference between offset + sats/usd?
         offset: 4,
+        //TODO: Is current offset 0.000001 to the dollar which would be 4 for offset and 2 for usd/cent?
+        // Why not use offset 6 and USD?
```
Comment (nicolasburtey) [inline src/graphql/types/abstract/transaction.ts:45]: the reason we are using cents and not dollar is to work around the limitation of graphql int. ie: to not overflow.

File: src/graphql/types/object/wallet-contact.ts (hunk 2 above)
Comment (nicolasburtey) [reply src/graphql/types/object/wallet-contact.ts]: good idea. use: ``

File: src/graphql/admin/schema.graphql (hunk 1 above)
Comment (krtk6160) [reply src/graphql/admin/schema.graphql:262]: `bitcoin layer` perhaps?

Comment (bodymindarts) [reply src/graphql/admin/schema.graphql:262]: Hmm its probably fine - can't think of something better.

File: src/graphql/types/object/wallet-contact.ts (hunk 2 above)
Comment (daviroo) [reply src/graphql/types/object/wallet-contact.ts]: 👍

File: src/graphql/types/object/user.ts (hunk 3 above)
Comment (daviroo) [reply src/graphql/types/object/user.ts]: I have switched out all strings which use new lines to use multi line strings.

**Review by samerbuna — APPROVED** [review]

File: src/graphql/types/abstract/transaction.ts [hunk 5]
```diff
@@ -25,11 +25,9 @@ export const transactionInterfaceFields = {
   },
   settlementAmount: {
     type: GT.NonNull(SatAmount),
-    //TODO: Does the amount include fees.
```
Comment (nicolasburtey) [inline src/graphql/types/abstract/transaction.ts]: this would still be relevant information to give to the user as part of the documentation

**Review by nicolasburtey — APPROVED** [review]

**Pull request merged**